src/commands/logs.rs

```diff
@@ -0,0 +1,120 @@
+    pub app_name: String,
+
+    /// Follow logs output
+    #[clap(name = "follow", long = "follow")]
+    pub follow: bool,
+
+    /// Number of lines to show from the end of the logs
+    #[clap(name = "tail", long = "tail", default_value = "10")]
+    pub lastn: usize,
+
+    /// interval in secs to refresh logs from cloud
+    #[clap(name = "interval", long = "interval", default_value = "5")]
+    pub interval_secs: u64,
+}
+
+impl LogsCommand {
+    pub async fn run(self) -> Result<()> {
+        let login_connection = login_connection(self.deployment_env_id.as_deref()).await?;
+        let app_name: String = self.app_name.clone();
+        self.tail_logs(login_connection, app_name)
+            .await
+            .map_err(|e| anyhow!("{:?}\n\nLearn more at {}", e, "DEVELOPER_CLOUD_FAQ"))
+    }
+
+    async fn tail_logs(self, login_connection: LoginConnection, app_name: String) -> Result<()> {
+        let connection_config = ConnectionConfig {
+            url: login_connection.url.to_string(),
+            insecure: login_connection.danger_accept_invalid_certs,
+            token: login_connection.token.clone(),
+        };
+
+        let client = CloudClient::new(connection_config.clone());
+        let app_id = get_app_id_cloud(&client, app_name.as_str()).await?;
+        let channel_id =
+            get_channel_id_cloud(&client, SPIN_DEPLOY_CHANNEL_NAME.to_string(), app_id).await?;
+        keep_printing_latest_logs(
+            &client,
+            channel_id,
+            self.follow,
+            self.interval_secs,
+            self.lastn,
+        )
+        .await?;
+
+        Ok(())
+    }
+}
+
+async fn keep_printing_latest_logs(
+    client: &CloudClient,
+    channel_id: Uuid,
+    follow: bool,
+    interval_secs: u64,
+    lastn: usize,
+) -> Result<()> {
+    let mut already_processed = fetch_and_print_lastn_logs(client, channel_id, lastn).await?;
+
+    if !follow {
+        return Ok(());
+    }
+
+    loop {
+        tokio::time::sleep(tokio::time::Duration::from_secs(interval_secs)).await;
+        already_processed =
+            fetch_and_print_latest_logs(client, channel_id, already_processed).await?;
+    }
+}
+
+async fn fetch_and_print_lastn_logs(
+    client: &CloudClient,
+    channel_id: Uuid,
+    lastn: usize,
+) -> Result<usize> {
+    let logs_resp = client.channel_logs(channel_id.to_string()).await?;
+    let count = logs_resp.logs.len();
+    print_lastn_logs(logs_resp.logs, lastn);
+    Ok(count)
+}
+
+async fn fetch_and_print_latest_logs(
+    client: &CloudClient,
+    channel_id: Uuid,
+    printed_so_far: usize,
+) -> Result<usize> {
+    let logs_resp = client.channel_logs(channel_id.to_string()).await?;
+    let count = logs_resp.logs.len();
+    print_lastn_logs(logs_resp.logs, count - printed_so_far);
+    Ok(count)
+}
+
+fn print_lastn_logs(logs: Vec<String>, n: usize) {
+    for log in logs.iter().rev().take(n) {
+        println!("{}", log)
+    }
+}
```

Comment:
I believe that this needs to be changes to take the last n and keep the direction of ordering. I was thinking through this in this playground https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=efb52801edab3b4baa36ea2e68499d96
```suggestion
fn print_lastn_logs(logs: Vec<String>, n: usize) {
    for log in &logs[logs.len()-n..] {
        println!("{}", log)
    }
}
```

Reply:
🎉  thank you.